aggregator/src/aggregator.rs

```diff
@@ -2086,7 +2111,7 @@ impl VdafOps {
                 Error::Internal("threadpool failed to send VDAF preparation result".into())
             })?
             .unwrap_or_else(|panic_cause: Box<dyn Any + Send>| {
-                panic_any(panic_cause);
+                resume_unwind(panic_cause);
```

Comment:
A note on this change: calling `panic_any` will result in a second panic, with all associated logging; `resume_unwind` will continue the "existing" panic without additional logic.

For example, the following program:

```rust
fn main() {
    println!("Before panic");
    let result = panic::catch_unwind(|| { panic!("uh oh") });
    println!("Before panic handling");
    if let Err(ongoing_panic) = result {
        panic::panic_any(ongoing_panic);
    }
    println!("After panic handling");
}
```

will print:

```
Before panic
thread 'main' panicked at src/main.rs:5:43:
uh oh
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Before panic handling
thread 'main' panicked at src/main.rs:8:9:
Box<dyn Any>
```

... while the following program:

```rust
fn main() {
    println!("Before panic");
    let result = panic::catch_unwind(|| { panic!("uh oh") });
    println!("Before panic handling");
    if let Err(ongoing_panic) = result {
        panic::resume_unwind(ongoing_panic);
    }
    println!("After panic handling");
}
```

will print:

```
Before panic
thread 'main' panicked at src/main.rs:5:43:
uh oh
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Before panic handling
```

Reply:
Thanks for figuring this out! The duplicate messages had bothered me previously, but I didn't get to the bottom of it.